packages/cloud_firestore/cloud_firestore/lib/src/firestore.dart

```diff
@@ -16,12 +16,14 @@ part of cloud_firestore;
 /// FirebaseFirestore firestore = FirebaseFirestore.instanceFor(app: secondaryApp);
 /// ```
 class FirebaseFirestore extends FirebasePluginPlatform {
-  FirebaseFirestore._({required this.app, required this.databaseURL})
-      : super(app.name, 'plugins.flutter.io/firebase_firestore') {
-    if (databaseURL.endsWith('/')) {
-      databaseURL = databaseURL.substring(0, databaseURL.length - 1);
-    }
-  }
+  FirebaseFirestore._({
+    required this.app,
+    @Deprecated(
+      '`databaseURL` has been deprecated. Please use `databaseId` instead.',
+    )
+    required this.databaseURL,
+    required this.databaseId,
+  }) : super(app.name, 'plugins.flutter.io/firebase_firestore');
```

Comment:
What happened to the following. Is it not needed anymore?

```dart
if (databaseURL.endsWith('/')) {
  databaseURL = databaseURL.substring(0, databaseURL.length - 1);
}
```

Reply:
no, it isn't a URL, it is a databaseId so it will be like this:
https://github.com/firebase/flutterfire/pull/12593/files#diff-69186946971119c262bbc33818abecdc9ae077adb87cdfd1c0a489787d6a7d44R40

The default databaseId is "(default)".